// reakit-theme-default/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var reakit = require('reakit');
var styledTools = require('styled-tools');

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

function _templateObject21() {
  var data = _taggedTemplateLiteralLoose(["\n  & .stroke {\n    fill: none;\n  }\n"]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _taggedTemplateLiteralLoose(["\n  text-transform: none;\n  pointer-events: none;\n  white-space: nowrap;\n  font-size: 0.875em;\n  text-align: center;\n  box-shadow: none;\n  border-radius: 0.25em;\n  padding: 0.75em 1em;\n"]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteralLoose(["\n  display: inline-flex;\n  position: relative;\n  flex: 1;\n  user-select: none;\n  outline: none;\n  align-items: center;\n  white-space: nowrap;\n  justify-content: center;\n  text-decoration: none;\n  height: 2.5em;\n  padding: 0 0.5em;\n  min-width: 2.5em;\n  &.active {\n    font-weight: bold;\n  }\n  &[disabled] {\n    pointer-events: none;\n  }\n"]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  align-items: center;\n  list-style: none;\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteralLoose(["\n  border: 1px solid ", ";\n  table-layout: fixed;\n  border-collapse: collapse;\n  background-color: ", ";\n  line-height: 200%;\n\n  tbody,\n  td,\n  th,\n  tfoot,\n  thead,\n  tr {\n    border: inherit;\n  }\n\n  caption {\n    text-transform: uppercase;\n    font-size: 0.9em;\n    color: ", ";\n  }\n\n  td,\n  th {\n    padding: 0 8px;\n    vertical-align: middle;\n  }\n\n  th {\n    font-weight: bold;\n    background-color: ", ";\n  }\n"]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteralLoose(["\n  border-radius: 0;\n"]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteralLoose(["\n  & .stroke {\n    fill: ", ";\n  }\n"]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteralLoose(["\n  padding: 1em;\n  border-radius: 0.25em;\n  box-shadow: 0 0 0 1px ", ", 0 2px 4px ", ",\n    0 8px 24px ", ";\n\n  &[aria-hidden=\"false\"] {\n    transition-timing-function: ", ";\n  }\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteralLoose(["\n  &:not(:last-child) {\n    margin-bottom: 1rem;\n  }\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteralLoose(["\n  padding: 1em;\n  border-radius: 0.25em;\n  box-shadow: 0 0 0 1px ", ", 0 4px 8px ", ",\n    0 16px 48px ", ";\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteralLoose(["\n  list-style: none;\n\n  li {\n    margin-bottom: 0.35em;\n  }\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteralLoose(["\n  display: inline-grid;\n  grid-gap: 0.25em;\n  align-items: center;\n  grid-auto-flow: column;\n  text-decoration: none;\n\n  &:hover {\n    text-decoration: underline;\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteralLoose(["\n  display: block;\n  width: 100%;\n  padding: 0 0.5em;\n  height: 2.5em;\n  border-radius: 0.25em;\n\n  &[type=\"checkbox\"],\n  &[type=\"radio\"] {\n    display: inline-block;\n    width: auto;\n    height: auto;\n    padding: 0;\n  }\n\n  &::placeholder {\n    color: currentcolor;\n    opacity: 0.5;\n  }\n\n  textarea & {\n    padding: 0.5em;\n    height: auto;\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteralLoose(["\n  display: block;\n  max-width: 100%;\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  font-weight: bold;\n  margin: 1em 0 0.5em;\n  &:first-child {\n    margin-top: 0;\n  }\n  h1& {\n    font-size: 2.5em;\n  }\n  h2& {\n    font-size: 2em;\n  }\n  h3& {\n    font-size: 1.75em;\n  }\n  h4& {\n    font-size: 1.5em;\n  }\n  h5& {\n    font-size: 1.25em;\n  }\n  h6& {\n    font-size: 1em;\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  border: 1px solid ", ";\n  border-radius: 0.25em;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  flex: 1;\n  label {\n    padding-bottom: 0.5em;\n  }\n  > *:not(label):not(:last-child) {\n    margin-bottom: 0.5em;\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  font-family: monospace;\n  white-space: pre-wrap;\n  word-wrap: break-word;\n  padding: ", ";\n  border-radius: 0.25em;\n\n  code {\n    display: block;\n    padding: 1em;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  display: inline-flex;\n  position: relative;\n  appearance: none;\n  align-items: center;\n  justify-content: center;\n  cursor: pointer;\n  min-width: 2.5em;\n  height: 2.5em;\n  padding: 0 0.68em;\n  border-radius: 0.25em;\n  flex: none;\n  user-select: none;\n  white-space: nowrap;\n  text-decoration: none;\n  outline: none;\n  &:hover,\n  &:focus {\n    box-shadow: inset 0 0 999em ", ";\n  }\n  &:active,\n  &.active {\n    box-shadow: inset 0 0 999em ", ";\n  }\n  &:after {\n    display: none;\n    content: \"\";\n    position: absolute;\n    top: 0;\n    right: 0;\n    bottom: 0;\n    left: 0;\n    border-radius: inherit;\n    background-color: rgba(255, 255, 255, 0.35);\n  }\n  &[disabled] {\n    pointer-events: none;\n    &:after {\n      display: block;\n    }\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  border-left: 5px solid ", ";\n  padding: 0.5em 0 0.5em 1em;\n  font-style: italic;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  border-radius: 50%;\n  width: 1.5em;\n  height: 1.5em;\n  overflow: hidden;\n  object-fit: cover;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var palette = {
  white: "#ffffff",
  whiteText: styledTools.palette("black"),
  black: "#212121",
  blackText: styledTools.palette("white"),
  // https://coolors.co/2196f3-42a5f5-64b5f6-90caf9-bbdefb
  primary: ["#2196f3", "#42a5f5", "#64b5f6", "#90caf9", "#bbdefb"],
  primaryText: [styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black")],
  // https://coolors.co/e91e63-ec407a-f06292-f48fb1-f8bbd0
  secondary: ["#e91e63", "#ec407a", "#f06292", "#f48fb1", "#f8bbd0"],
  secondaryText: [styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black")],
  // https://coolors.co/f44336-ef5350-e57373-ef9a9a-ffcdd2
  danger: ["#f44336", "#ef5350", "#e57373", "#ef9a9a", "#ffcdd2"],
  dangerText: [styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black")],
  // https://coolors.co/ffc107-ffca28-ffd54f-ffe082-ffecb3
  alert: ["#ffc107", "#ffca28", "#ffd54f", "#ffe082", "#ffecb3"],
  alertText: [styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black")],
  // https://coolors.co/4caf50-66bb6a-81c784-a5d6a7-c8e6c9
  success: ["#4caf50", "#66bb6a", "#81c784", "#a5d6a7", "#c8e6c9"],
  successText: [styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("black"), styledTools.palette("black")],
  grayscale: [styledTools.palette("black"), "#414141", "#616161", "#9e9e9e", "#bdbdbd", "#e0e0e0", "#f5f5f5", styledTools.palette("white")],
  grayscaleText: [styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black")],
  background: [styledTools.palette("grayscale", -4), styledTools.palette("grayscale", -3), styledTools.palette("grayscale", -2), styledTools.palette("grayscale", -1)],
  backgroundText: styledTools.palette("black"),
  shadow: ["rgba(0, 0, 0, 0.9)", "rgba(0, 0, 0, 0.7)", "rgba(0, 0, 0, 0.5)", "rgba(0, 0, 0, 0.3)", "rgba(0, 0, 0, 0.15)", "rgba(0, 0, 0, 0.075)"],
  shadowText: [styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("white"), styledTools.palette("black"), styledTools.palette("black"), styledTools.palette("black")],
  transparent: "transparent",
  transparentText: styledTools.palette("black"),
  border: styledTools.palette("shadow", -2)
};
var Avatar = reakit.css(_templateObject());
var Blockquote = reakit.css(_templateObject2(), styledTools.palette("grayscale", -3));
var Button = reakit.css(_templateObject3(), styledTools.palette("shadow", -2), styledTools.palette("shadow", -3));
var Code = reakit.css(_templateObject4(), styledTools.ifProp("block", "0", "0.25em 0.35em"));
var Field = reakit.css(_templateObject5());
var GroupItem = reakit.css(_templateObject6(), styledTools.palette("border"));
var Heading = reakit.css(_templateObject7());
var Image = reakit.css(_templateObject8());
var Input = reakit.css(_templateObject9());
var Link = reakit.css(_templateObject10());
var List = reakit.css(_templateObject11());
var Overlay = reakit.css(_templateObject12(), styledTools.palette("shadow", -2), styledTools.palette("shadow", -2), styledTools.palette("shadow", -2));
var Paragraph = reakit.css(_templateObject13());
var Popover = reakit.css(_templateObject14(), styledTools.palette("shadow", -2), styledTools.palette("shadow", -1), styledTools.palette("shadow", -1), styledTools.prop("timing", "cubic-bezier(0.25, 0.1, 0.25, 1.5)"));
var PopoverArrow = reakit.css(_templateObject15(), styledTools.palette("shadow", -2));
var Sidebar = reakit.css(_templateObject16());
var Table = reakit.css(_templateObject17(), styledTools.palette("grayscale", 4), styledTools.palette("background", -1), styledTools.palette("grayscale", 3), styledTools.palette("shadow", -1));
var Tabs = reakit.css(_templateObject18());
var TabsTab = reakit.css(_templateObject19());
var Tooltip = reakit.css(_templateObject20());
var TooltipArrow = reakit.css(_templateObject21());
var index = {
  palette: palette,
  Avatar: Avatar,
  Blockquote: Blockquote,
  Button: Button,
  Code: Code,
  Field: Field,
  GroupItem: GroupItem,
  Heading: Heading,
  Image: Image,
  Input: Input,
  Link: Link,
  List: List,
  Overlay: Overlay,
  Paragraph: Paragraph,
  Popover: Popover,
  PopoverArrow: PopoverArrow,
  Sidebar: Sidebar,
  Table: Table,
  Tabs: Tabs,
  TabsTab: TabsTab,
  Tooltip: Tooltip,
  TooltipArrow: TooltipArrow
};

exports.palette = palette;
exports.Avatar = Avatar;
exports.Blockquote = Blockquote;
exports.Button = Button;
exports.Code = Code;
exports.Field = Field;
exports.GroupItem = GroupItem;
exports.Heading = Heading;
exports.Image = Image;
exports.Input = Input;
exports.Link = Link;
exports.List = List;
exports.Overlay = Overlay;
exports.Paragraph = Paragraph;
exports.Popover = Popover;
exports.PopoverArrow = PopoverArrow;
exports.Sidebar = Sidebar;
exports.Table = Table;
exports.Tabs = Tabs;
exports.TabsTab = TabsTab;
exports.Tooltip = Tooltip;
exports.TooltipArrow = TooltipArrow;
exports.default = index;
